runtime/reconcilers/model.go

```diff
@@ -1233,20 +1234,48 @@ func (r *ModelReconciler) executeSingle(ctx context.Context, executor *wrappedMo
 		return nil, err
 	}
 
-	// Execute the stage step if configured
-	var stageDuration time.Duration
-	if executor.stage != nil {
-		// Also resolve templating in the stage props
-		stageProps, err := r.resolveTemplatedProps(ctx, self, incrementalState, partitionData, mdl.Spec.StageConnector, mdl.Spec.StageProperties.AsMap())
-		if err != nil {
-			return nil, err
+	return r.executeWithRetry(ctx, self, mdl, func(ctx context.Context) (*drivers.ModelResult, error) {
+		var finalResult *drivers.ModelResult
```

Comment:
There doesn't seem to be any reason to declare this separately, can just use `:=` when it's needed like the old code. 

I also notice many comments are slightly changed. Is it AI edits? It's kind of annoying to review with all these small edits in code that the PR doesn't really need to change.

Reply:
Agreed, I will be mindful of that.